Dockerfile

```diff
@@ -0,0 +1,45 @@
+#
+# Copyright 2021 The Sigstore Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM golang:1.19.3@sha256:7ffa70183b7596e6bc1b78c132dbba9a6e05a26cd30eaa9832fecad64b83f029 as builder
+ENV APP_ROOT=/opt/app-root
+ENV GOPATH=$APP_ROOT
+
+WORKDIR $APP_ROOT/src/
+ADD go.mod go.sum $APP_ROOT/src/
+RUN go mod download
+
+# Add source code
+ADD ./cmd/ $APP_ROOT/src/cmd/
+ADD ./pkg/ $APP_ROOT/src/pkg/
+
+RUN go build ./cmd/mirroring
+RUN CGO_ENABLED=0 go build -gcflags "all=-N -l"  -o mirroring_debug ./cmd/mirroring
+
+# Multi-Stage build
+FROM golang:1.19.3@sha256:7ffa70183b7596e6bc1b78c132dbba9a6e05a26cd30eaa9832fecad64b83f029 as deploy
+
+# Retrieve the binary from the previous stage
+COPY --from=builder /opt/app-root/src/mirroring /usr/local/bin/mirroring
+
+# Set the binary as the entrypoint of the container
+CMD ["mirroring"]
+
+# debug compile options & debugger
+FROM deploy as debug
```

Comment:
there is no reference to this debug container in use anywhere. Do we expect this to be used by users of the monitor? or just by developers/contributors to the codebase?

Reply:
I'm not too familiar with how to construct dockerfiles, and this was mostly copy-paste from other examples, but I thought this would include the dlv debugger in the container? since this is a multi stage build, it would be the debug container that gets deployed

Reply:
well it's creating multiple targets. You reference the `deploy` target in your docker-compose.yaml, but presumably this doesn't matter.

Reply:
https://docs.docker.com/build/building/multi-stage/#stop-at-a-specific-build-stage

Reply:
anyway, this doesn't matter. Leaving it in is fine.

Reply:
oh that's what target is meant for. thanks! i'll test out locally to see if debug is even necessary